### ISSUE TITLE
Bug report: not entering into iterator until thorough depth.

### ISSUE THREAD

**Issue by neuralix** [torchnet/torchnet#76]
In this case,
```
local function getIterator(mode)
   return tnt.ParallelDatasetIterator{
         return tnt.BatchDataset{
            dataset = tnt.ListDataset{
                [place 1]
```

When I set batchSize as big one(:eg. 1000), 
I met a case, one cannot reache the [place 1].

But strangely, the program runs without any error.
And it generating 'nan' training loss.

**Comment by lvdmaaten** [comment]
I don't see how your code example can work at all. `tnt.ParallelDatasetIterator` expects a `closure` as input, which it does not get here. The closure is supposed to return the dataset.

**Comment by lvdmaaten** [comment]
Closing because of insufficient details on issue.
